test/distributed/test_store.py

```diff
@@ -0,0 +1,425 @@
+import os
+import random
+import sys
+import tempfile
+import time
+import traceback
+import unittest
+from datetime import timedelta
+from sys import platform
+
+import torch
+import torch.distributed as c10d
```

Comment:
nit: (as you mentioned, this is prior to this PR) Shall we add a followup PR to consolidate c10d and dist?